pkg/reconciler.v1/common/job_must_customize.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+
+	commonv1 "github.com/kubeflow/common/pkg/apis/common/v1"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetJob returns the job that matches the request
+func (r *KubeflowJobReconciler) GetJob(ctx context.Context, req ctrl.Request) (client.Object, error) {
+	panic("implement KubeflowReconciler.GetJob!")
```

Comment:
Does it need to implement all methods? 

These methods will be overridden by upstream reconciler? is there any chance to call these and get panic?

Reply:
At this stage, it is designed to call the panic if developers uses `KubeflowJobReconciler` directly. It forces developers to override these methods.

However, after the generic `KubeflowJob` is introduced in tf-operator or we call unified training operator, these `panic` functions will be changed into warning or logging info.

Reply:
If you forces developer to override these methods. Why just not to implement these methods so end user will get compilation error. This way gives user runtime error. is that correct?

Reply:
I see, exactly! Now all these `panic` methods are removed. Meanwhile, the `job_must_customize.go` file is replaced by a `README.md` to inform developers these function should be implemented. A simple example can be found at `test_job/reconciler.v1`

Reply:
Thanks for the update. Great!